sdk/identity/azure-identity/src/main/java/com/azure/identity/ManagedIdentityServiceCredential.java

```diff
@@ -0,0 +1,49 @@
+package com.azure.identity;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.implementation.IdentityClient;
+import reactor.core.publisher.Mono;
+
+/**
+ * The Managed Service Identity credential.
+ */
+abstract class ManagedIdentityServiceCredential {
+    final IdentityClient identityClient;
+    final String clientId;
```

Comment:
should `clientId` and `environment` be private since they have getters?

Reply:
made private.